v1/modules/ble/src/ble_instr_utils.c

```diff
@@ -203,53 +204,108 @@ VECTOR_HANDLE parse_instructions(JSON_Array* instructions)
+                            JSON_Array* seq_instructions = json_object_get_array(instr, "instructions");
+                            VECTOR_HANDLE seqInsts = parse_instructions(seq_instructions);
+                            if (seqInsts!=NULL)
+                            {
+                                size_t numOfSeqs = VECTOR_size(seqInsts);
+                                if (numOfSeqs>0)
+                                {
+  									BLE_INSTRUCTION* prev_instr = (BLE_INSTRUCTION*)VECTOR_element(seqInsts, 0);
+									BLE_INSTRUCTION firstInstr = { 0 };
+									firstInstr.instruction_type = prev_instr->instruction_type;
+									firstInstr.characteristic_uuid = STRING_clone(prev_instr->characteristic_uuid);
+									firstInstr.nextInst = NULL;
+									memcpy(&(firstInstr.data), &(prev_instr->data), sizeof(prev_instr->data));
+									prev_instr = &firstInstr;
+									if (numOfSeqs > 1)
+                                    {
+										for (size_t c = 1; c < numOfSeqs; c++)
+                                        {
+											BLE_INSTRUCTION* current = (BLE_INSTRUCTION*)VECTOR_element(seqInsts, c);
+                                            BLE_INSTRUCTION* currentInstr = (BLE_INSTRUCTION*)malloc(sizeof(BLE_INSTRUCTION));
+                                            if (currentInstr==NULL)
+                                            {
+                                                LogError("malloc return NULL for new currentInstr memory allocation.");
+                                            }
+                                            else
+                                            {
+    											currentInstr->instruction_type = current->instruction_type;
+                                                currentInstr->characteristic_uuid = STRING_clone(current->characteristic_uuid);
+                                                if (currentInstr->characteristic_uuid==NULL)
+                                                {
+                                                    LogError("STRING_clone return NULL for current->characteristic_uuid clone.");
+                                                    free(currentInstr);
+                                                }
+                                                else
+                                                {
+    		    									memcpy(&(currentInstr->data), &(current->data), sizeof(current->data));
+	    		    								prev_instr->nextInst = currentInstr;
+                                                    prev_instr = currentInstr;
+                                                }
+                                            }
+										}
+									}
+									if (VECTOR_push_back(result, &firstInstr, 1) != 0)
+                                    {
+                                        LogError("VECTOR_push_back return not zero for the property 'characteristic_uuid' for instruction number %zu", i);
```

Comment:
I think I had the same comment on the old PR - would it make sense to continue if this fails?